### PR TITLE
Enhance build system documentation

### PR DESCRIPTION
**Pull request by atharvalele**
- Add details on OsLoader and FW update packaging
- Add section on post build image construction
- Add section on patching of stages
- Add section on including SBL containers in final SBL image

### CODE REVIEW

File: source/developer-guides/build-system.rst [hunk 1]
```diff
@@ -90,15 +104,170 @@ Slim Bootloader is built in stages and more information on each stage is given b
...
+  of the build process.
+* The final image layout for Slim Bootloader can be defined in ``BoardConfig.py`` - ``GetImageLayout()``
+
+  * If firmware resiliency is enabled, the layout will typically contain redundant copies of the boot-critical code.
+    Check :ref:`firmware-resiliency-and-recovery` for more details.
+
+* As discussed above, the final SBL image includes SBL stages and other components, examples of which are listed below:
+
+  * These additional components included in the SBL image are placed in the non-redundant region of the image
+  * The list of included containers or components in the final image is present in the ``GetImageLayout()`` function
```
Comment (rprangar) [inline source/developer-guides/build-system.rst:134]: Can you remove callout to containers here?

File: source/developer-guides/build-system.rst [hunk 2]
```diff
@@ -90,15 +104,170 @@ Slim Bootloader is built in stages and more information on each stage is given b
...
+    * ``MRCDATA.bin`` - Memory Reference Code Data, used to store Memory Reference Code training data
+    * ``VARIABLE.bin`` - This region is used for ``GetVariable()`` and ``SetVariable()`` APIs provided by ``LiteVariableLib`` in ``BootloaderCommonPkg``
+    * ``UCODE.bin`` - CPU Microcode
+
+    * The components listed above are generated by the ``post_build()`` function in ``BuildLoader.py``
+
+  * Payloads other than OsLoader and FW Update are built independently and the payload binary is packaged as ``EPAYLOAD.bin``
+    to be included in the final SBL image
+
+    * ``EPAYLOAD.bin`` - Payload for SBL - used for extra payloads (UEFI Payload, Linux, u-boot, etc.)
```
Comment (rprangar) [inline source/developer-guides/build-system.rst:147]: Please add a reference to - https://slimbootloader.github.io/how-tos/integrate-multiple-payloads.html#integrate-multiple-payloads

File: source/developer-guides/build-system.rst (hunk 1 above)
Comment (rprangar) [inline source/developer-guides/build-system.rst:134]: The list of included containers or components in the final image --> The list of components in the final image

File: source/developer-guides/build-system.rst (hunk 2 above)
Comment (atharvalele) [reply source/developer-guides/build-system.rst]: Added, thanks

File: source/developer-guides/build-system.rst (hunk 1 above)
Comment (atharvalele) [reply source/developer-guides/build-system.rst]: Thanks, changed it

**Review by rprangar — COMMENTED** [review]
LGTM

**Review by rprangar — APPROVED** [review]
LGTM

**Pull request merged**